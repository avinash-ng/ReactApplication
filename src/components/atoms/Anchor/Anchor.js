import React from 'react'

const Anchor = (props) => {
    return (
        <a href={props.href}>{props.name}</a>
    )
}
export default Anchor