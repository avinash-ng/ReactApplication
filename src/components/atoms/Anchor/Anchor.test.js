import Anchor from './Anchor'
import expect from 'expect'
import { shallow, configure } from 'enzyme'
import React from 'react'
import Adapter from 'enzyme-adapter-react-16';
// import renderer from 'react-test-renderer'
import { render } from '@testing-library/react';

configure({ adapter: new Adapter() });
describe('<Anchor /> ', () => {
    it('renders anchor element', () => {
        
        const RenderComponent = shallow(
            <Anchor ></Anchor>
        );
        // console.log(RenderComponent)
        expect(RenderComponent.exists()).toBe(true)
    })

    it('recieved props correctly', () => {
        const name = "click me";
        const ref = "https://www.facebook.com"
        const ref1 = "https://www.google.com"
        const { getByText } = render(<Anchor name={name} href={ref}></Anchor>)
        // console.log(RenderComponent)
        expect(getByText('click me').closest("a")).toHaveAttribute('href', ref)
    })


    it('Matches snapshot', () => {
        const name = "click me";
        const ref = "https://www.facebook.com"
        const render = shallow(<Anchor name={name} href={ref} ></Anchor>)
        expect(render).toMatchSnapshot()
    })
})