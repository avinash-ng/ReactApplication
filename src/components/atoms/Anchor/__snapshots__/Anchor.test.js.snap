// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Anchor />  Matches snapshot 1`] = `ShallowWrapper {}`;
